src/pytorch_lightning/trainer/connectors/signal_connector.py

```diff
@@ -37,17 +37,15 @@ def __call__(self, signum: _SIGNUM, frame: FrameType) -> None:
 class SignalConnector:
     def __init__(self, trainer: "pl.Trainer") -> None:
         self.trainer = trainer
-        self.trainer._terminate_gracefully = False
+        self.trainer._received_sigterm = False
         self._original_handlers: Dict[_SIGNUM, _HANDLER] = {}
 
     def register_signal_handlers(self) -> None:
+        self.trainer._received_sigterm = False
         self._original_handlers = self._get_current_signal_handlers()
 
         sigusr_handlers: List[_HANDLER] = []
-        sigterm_handlers: List[_HANDLER] = []
-
-        if _fault_tolerant_training():
-            sigterm_handlers.append(self.fault_tolerant_sigterm_handler_fn)
+        sigterm_handlers: List[_HANDLER] = [self._sigterm_notifier_fn]
```

Comment:
This has the implication that the SIGTERM signal will be ignored until the loops reach a point that check for it. After that, they will raise the `ExitGracefullyException`